core/style.js

```diff
@@ -2117,3 +2117,81 @@ CKEDITOR.tools.extend( CKEDITOR.editor.prototype, {
  * @cfg {String/Array/Boolean} [stylesSet='default']
  * @member CKEDITOR.config
  */
+
+/**
+ * Abstract class describing definition of a style.
+ * It is used as one of {@link CKEDITOR.style#constructor}'s arguments.
+ *
+ * Style definition object represents a style as a set of properties describing CSS style rules and HTML attributes.
```

Comment:
> (…) properties describing CSS style rules and HTML attributes.

We lost one information here - it could also store element type, which is neither HTML attr nor is it a CSS style.

Though I don't have a clear idea how to write it here 😜

Reply:
Maybe something like:

> Style definition object represents a style as a set of properties describing CSS style rules, HTML attributes, style element and name.

or maybe more general one:

> Style definition object represents a style as a set of properties defining element structure, its attributes and styles/CSS properties.

Reply:
And then we could skip next sentence

> It also store element type which can be used to provide valid HTML element as a wrapper or replacement for a selection's HTML.

as it seems a little redundant.